app/assets/javascript/controllers/multiple_select_component_controller.js

```diff
@@ -85,4 +91,40 @@ export default class extends SelectComponentController {
       },
     );
   }
+
+  #truncateItemsAddlisteners(selectize) {
+    this.#truncateItems(selectize);
+
+    selectize.on("change", () => {
+      this.#truncateItems(selectize);
+    });
+
+    window.addEventListener("resize", () => {
+      this.#truncateItems(selectize);
+    });
+  }
+
+  #truncateItems(selectize) {
+    const $selectizeInput = selectize.$control;
+    const inputWidth = $selectizeInput[0].clientWidth -
+      parseFloat($selectizeInput.css("padding-left")) -
+      parseFloat($selectizeInput.css("padding-right"));
+
+    const [firstItem, ...otherItems] = $selectizeInput.find(".item");
+
+    if (!firstItem) return;
+
+    const firstItemWidth = firstItem.clientWidth;
+
+    if (inputWidth <= firstItemWidth) {
+      $selectizeInput.css("text-overflow", "");
+      otherItems.forEach((item) => item.style.visibility = "hidden");
+    } else if (selectize.items.length > 1) {
+      $selectizeInput.css("text-overflow", "ellipsis");
+      [firstItem, ...otherItems].forEach((item) => item.style.visibility = "");
+    } else {
+      $selectizeInput.css("text-overflow", "");
+      [firstItem, ...otherItems].forEach((item) => item.style.visibility = "");
```

Comment:
difference than in internal project, to fix mistake with invisible first item.

how to repeat in the internal project:
 - select long name
 - select short name that will be invisible
 - deselect long name
 - short name will be invisible

Reply:
Would we like to fix this in an internal project?

Reply:
will create an issue

Reply:
[created pr](https://github.com/toughbyte/hub/pull/15900)